presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -538,4 +548,28 @@ default void revokeTablePrivileges(ConnectorSession session, SchemaTableName tab
     {
         return emptyList();
     }
+
+    /**
+     * Whether the connector uses the legacy Table Layout feature. If this method returns false,
+     * connectors are required to implement the following methods:
+     * <ul>
+     * <li>{@link #getTableProperties(ConnectorSession session, ConnectorTableHandle table)}</li>
+     * <li>{@link #getInfo(ConnectorTableHandle table)} </li>
+     * <li>{@link ConnectorSplitManager#getSplits(ConnectorTransactionHandle, ConnectorSession, ConnectorTableHandle, ConnectorSplitManager.SplitSchedulingStrategy)}</li>
+     * </ul>
+     */
+    default boolean usesLegacyTableLayouts()
+    {
+        return true;
+    }
+
+    default ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
+    {
+        if (!usesLegacyTableLayouts()) {
+            throw new IllegalStateException("getTableProperties() must be implemented if usesLegacyTableLayouts is false");
+        }
+
+        // TODO: ensure single layout
```

Comment:
maybe do this?

Reply:
Ah, yes, forgot to fix that before I submitted the PR